### PR TITLE
make SERVICE_ID column as FK in SERVICE_ORDER table

### LINKED ISSUE #1977: make SERVICE_ID column as FK in SERVICE_ORDER table
- [ ] Make SERVICE_ID column as a foreign key from DEPLOY_SERIVCE table. 
- [ ] Ensure purging of a service also purges the SERVICE_ORDER table entries for that service ID

### PR DESCRIPTION
**Pull request by WangLiNaruto**
Fixed #1977 
image

image
image
image

image
image

image
image
image

image
image

image
image
image

image
image

image
image
image

### CODE REVIEW

**Comment by baixinsui** [comment]
LGTM

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceOrderManager.java [hunk 1]
```diff
@@ -178,7 +186,7 @@ public ServiceOrderDetails getOrderDetailsByOrderId(UUID orderId) {
      */
     public ServiceOrderDetails getLatestModificationOrder(UUID serviceId) {
         ServiceOrderEntity query = new ServiceOrderEntity();
-        query.setServiceId(serviceId);
+        query.setDeployServiceEntity(deployServiceStorage.findDeployServiceById(serviceId));
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceOrderManager.java:189]: do we really need to set the full object in query?
probably it is enough to set `query.setServiceId(deployServiceStorage.findDeployServiceById(serviceId).getId());`

File: modules/database/src/main/java/org/eclipse/xpanse/modules/database/serviceorder/DatabaseServiceOrderStorage.java [hunk 2]
```diff
@@ -47,9 +47,9 @@ public List<ServiceOrderEntity> queryEntities(
         Specification<ServiceOrderEntity> specification =
                 (root, query, criteriaBuilder) -> {
                     List<Predicate> predicateList = new ArrayList<>();
-                    if (Objects.nonNull(entity.getServiceId())) {
-                        predicateList.add(criteriaBuilder.equal(root.get("serviceId"),
-                                entity.getServiceId()));
+                    if (Objects.nonNull(entity.getDeployServiceEntity())) {
+                        predicateList.add(criteriaBuilder.equal(root.get("deployServiceEntity"),
+                                entity.getDeployServiceEntity()));
```
Comment (swaroopar) [inline modules/database/src/main/java/org/eclipse/xpanse/modules/database/serviceorder/DatabaseServiceOrderStorage.java:52]: same here. Query should be based on same Id. not the complete entity.

**Review by swaroopar — CHANGES_REQUESTED** [review]

**Review by baixinsui — COMMENTED** [review]
are these modified tables available in both the H2 and MySQL databases? This is important.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**